swim/memberlist.go

```diff
@@ -356,6 +386,11 @@ func (m *memberlist) Apply(change Change) {
 	member, ok := m.members.byAddress[change.Address]
 
 	if !ok {
+		// avoid indefinite tombstones by not creating new nodes
+		// directly in this state
```

Comment:
I can't find a code path that leads to this code. Could you explain why this is necessary?